single-kernel/nbody.cpp

```diff
@@ -105,41 +105,19 @@ class NBody
       host_resulting_velocities[i] = new_v;
     }
 
-    double deviation = std::sqrt(
-        calculateSquaredDifference(host_resulting_particles.data(), resulting_particles.get_pointer(), particles.size()) +
-        calculateSquaredDifference(host_resulting_velocities.data(), resulting_velocities.get_pointer(), particles.size()));
-
-    return deviation < 1.e-6;
+    const float_type maxErr = std::sqrt(std::numeric_limits<float_type>::epsilon());
```

Comment:
Is there a mathematical reason for taking the √ here?

Reply:
Not really, just to obtain a larger constant :shrug:

Reply:
Well, I guess the `1.e-6` was also determined empirically 😉

I think your proposed approach for calculating the error is much more concise, but if we were to just raise the error bound for the original computation, how much greater than `1.e-6` would that have to be?

Reply:
We're getting around `1e-12` for `double` and `1e-3` for `float` IIRC

Reply:
I updated the description. This is failing due to optimizations performed by the driver. I think the new verification is a bit more forgiving for this kind of optimizations (while still requiring high precision for `float`) and does not depend on the number of inputs, so that's also an improvement.